### ISSUE TITLE
[clippy::doc_markdown] Default whitelist: Add 'WebAssembly'

### ISSUE THREAD

**Issue by Starwort** [rust-lang/rust-clippy#10998]
### Summary

Given that Rust is the most prominent WebAssembly language, it would make sense to add WebAssembly to the default list of `doc-valid-idents` (more sense, IMO, than ClojureScript or CoffeeScript!)

### Lint Name

clippy::doc_markdown

### Reproducer

I tried this code:

```rust
//! Yada yada this project is useful if you're working with WebAssembly yada yada
```

I saw this happen:

> item in documentation is missing backticks  
> for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown  
> try: ``WebAssembly``

I expected to see this happen:

No lint, as WebAssembly is a fairly common term, is the correct capitalisation of the standard, and is particularly pertinent to Rust

### Version

```text
rustc 1.72.0-nightly (498553fc0 2023-05-29)
binary: rustc
commit-hash: 498553fc04f6a3fdc53412320f4e913bc53bc267
commit-date: 2023-05-29
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.4
```
